st2reactor/st2reactor/rules/filter.py

```diff
@@ -149,7 +154,36 @@ def _render_criteria_pattern(self, criteria_pattern):
             # makes no sense
             return criteria_pattern
 
-        criteria_pattern = render_template_with_system_context(value=criteria_pattern)
+        LOG.debug(
+            'Rendering criteria pattern (%s) with context: %s',
+            criteria_pattern,
+            criteria_context
+        )
+
+        to_complex = False
+
+        if len(re.findall(r'{{[A-z0-9_.-]+}}', criteria_pattern)) > 0:
```

Comment:
Also, what if there are spaces in between the curly braces (i.e. `{{ foobar }}`)?